src/test/java/Homework22/BasePage22.java

```diff
@@ -0,0 +1,33 @@
+package Homework22;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import java.time.Duration;
+
+public class BasePage22 {
+
+    WebDriver driver;
+    WebDriverWait wait;
+    Actions actions;
```

Comment:
I recommend: make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

For example

```
    protected WebDriver driver;
    protected WebDriverWait wait;
    protected Actions actions;
```

Reply:
We haven't learned about using protected but thanks so much for the recommendation. I'll have to look into this!